types/underscore/underscore-tests.ts

```diff
@@ -1,3 +1,324 @@
+// common testing types and objects
+interface SimpleStringObject {
+    a: string;
+    b: string;
+}
+
+const simpleStringObjectArray: SimpleStringObject[] = [{ a: 'a', b: 'c' }, { a: 'b', b: 'b' }, { a: 'c', b: 'a' }];
+const simpleStringObjectList: _.List<SimpleStringObject> = { 0: { a: 'a', b: 'c' }, 1: { a: 'b', b: 'b' }, 2: { a: 'c', b: 'a' }, length: 3 };
+
+const simpleStringObjectDictionary: _.Dictionary<SimpleStringObject> = { a: { a: 'a', b: 'c' }, b: { a: 'b', b: 'b' }, c: { a: 'c', b: 'a' } };
+
+const simpleString = 'abc';
+
+const simpleNumber = 7;
+
+// Arrays
+
+// chunk
+{
+    {
+        const length = 2;
+        let result: SimpleStringObject[][];
+
+        // $ExpectType SimpleStringObject[][]
+        result = _.chunk<SimpleStringObject>(simpleStringObjectArray, length);
```

Comment:
Why test function calls with explicit type parameters? Personally, I think it is much more interesting whether TS will do the right thing without any type annotations.

Reply:
I agree that testing inference is definitely the more interesting case since I wouldn't often expect folks to specify types, but since types can be specified it seemed like the tests would be more complete if that case was tested as well. I'm fine with removing these, though, if you think they're overkill.

Reply:
It's not a strong opinion, but I feel that if typechecking is correct with automatic type inference, then it is probably also correct with explicit annotation. The other way round is not true.

Reply:
Sounds good to me, I like the idea of less lines in general and updating the generics in some of my experiments around possible future changes was a bit of a nightmare that I'd be happy to not have again 😄

Reply:
I've removed the duplicate tests that explicitly specify generics and also moved $ExpectType to the end of each line instead of putting it above them, which may help alleviate your concerns around the extra length that they were adding.